### PR TITLE
add scenario for handling tracking w/o CCC

### PR DESCRIPTION
**Pull request by mtosi**
as suggested by @mmusich 
I created a new scenario for handling the possibility having track reconstruction w/o CCC run @tier0

@rovere @VinInn @dimattia @boudoul @mmusich

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: Configuration/DataProcessing/python/Impl/trackingRun2noCCC.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+"""
+_hcalnzsRun2_
```
Comment (mmusich) [inline Configuration/DataProcessing/python/Impl/trackingRun2noCCC.py]: @mtosi this one doesn't look appropriate...

**Comment by mtosi** [comment]
the implementation is closer to a trick, rather than a proper handling of each instances of the cut w/in the reco config,
but I was not able to handle all different instances :(

**Comment by boudoul** [comment]
Hi @mtosi , Sorry for the naive question, could you please remind me the purpose of this ? Thanks

**Comment by slava77** [comment]
hold

There are a few issues:
- what's the point of tracking-only scenario ( ``` recoSeq=':reconstruction_trackingOnly' ``` )
- please use era-based scenarios
- the implementation should be based on a function that can be applied with cmsDriver (either define a new Era (most appropriate) or use a customization function, which can be in RecoTLR or elsewhere). 
- add the new scenario to the run_cfgTest.sh 
- minor: please do not use (at)tier0 in github: there is a user outside CMS

**Comment by slava77** [comment]
Same as question from @boudoul : 
is this intended for a specialized stream/dataset processing?
As defined, it can't be used for the main physics processing.

**Comment by davidlange6** [comment]
Lets follow up in a reco meeting (next week?) - the bulk (any process object manipulation) of such a feature should not be in Configuration/DataProcessing in any case.

**Comment by mtosi** [comment]
sorry,
this is meant for having the possibility of running the tracking w/o CCC on dedicated streams/PDs in case of need

**Comment by slava77** [comment]
Mia,
I suggest you use Configuration/DataProcessing/python/Impl/hcalnzsEra_Run2_2016.py as a reference and define trackingnocccEra_Run2_2016.py 
The method to turn off CCC should be taken from a noCCC era (to be defined) or from a customization function; either way, actual implementation of the methods disabling CCC  should be  outside of the Configuration/DataProcessing, DataProcessing should just call what's otherwise callable by cmsDriver.

**Comment by mtosi** [comment]
I did this update using CMSSW_8_0_0
and as example,
I made use of Configuration/DataProcessing/python/Impl/hcalnzsRun2.py

I should move to a most recent release ?
thanks

**Comment by slava77** [comment]
On 2/24/16 7:32 AM, mia tosi wrote:
> I did this update using CMSSW_8_0_0
> and as example,
> I made use of Configuration/DataProcessing/python/Impl/hcalnzsRun2.py
> 
> I should move to a most recent release ?

Yes, please.
800patch1 has the feature I was talking about.
Sorry, I got to implementation of this late in the cycle.

**Comment by mtosi** [comment]
thanks for pointing to it
I see it indeed in one of the most recent IB (81x)
ok, so I'll close this PR, implement this update in 800p1 and do a new PR

**Comment by mtosi** [comment]
sorry
but I have a question
Configuration/DataProcessing/python/Impl/hcalnzsEra_Run2_2016.py  is calling
Configuration/DataProcessing/python/Impl/hcalnzs.py 

where the last one is almost the same as the one I created in this branch for the tracking
if I understand properly your instructions, I do not have to overwrite the CCC parameters here
but I should create a new customization function (let say, RecoTracker/Configuration/python/customiseForRunIInoCCC.py) where I modify the parameters
and which is called by Configuration/DataProcessing/python/trackingnoccc.py
right ?
doing something like
args['customs'].append('RecoTracker.Configuration.customiseForRunIInoCCC') ?

thanks

**Comment by slava77** [comment]
On 2/24/16 9:03 AM, mia tosi wrote:
> sorry
> but I have a question
> Configuration/DataProcessing/python/Impl/hcalnzsEra_Run2_2016.py is calling
> Configuration/DataProcessing/python/Impl/hcalnzs.py
> 
> where the last one is almost the same as the one I created in this
> branch for the tracking
> if I understand properly your instructions, I do not have to overwrite
> the CCC parameters here
> but I should create a new customization function (let say,
> RecoTracker/Configuration/python/customiseForRunIInoCCC.py) where I
> modify the parameters
> and which is called by Configuration/DataProcessing/python/trackingnoccc.py
> right ?
> doing something like
> args['customs'].append('RecoTracker.Configuration.customiseForRunIInoCCC') ?

It is going to be more complete to start from "class trackingnoccc(pp)"
and have the noccc matters implemented in this base class.
Especially if this trackingnoccc is expected to stay for a while.

instead of args['custom'], I suggest you extend the corresponding
self.promptCustoms, etc.
(see/compare constructors in pp.py vs ppEra_Run2_2016).

**Comment by mtosi** [comment]
before proceeding w/ a new PR (triggering all the tests and waste of time)
could you please give a look at what I did in the following branch ?
https://github.com/cms-sw/cmssw/compare/CMSSW_8_1_X...mtosi:from-CMSSW_8_0_0_patch1_trackingWOccc

I hope I have understood your suggestions in the proper way
if there are still weird/silly points, please let me know

thanks much

**Comment by slava77** [comment]
Mia,

As noted in private emails, this looks in line with my suggestions/comments.
Please go ahead with a PR (update this or make anew)

**Pull request closed without merging**